prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -225,29 +225,46 @@ int efa_rdm_ep_post_user_recv_buf(struct efa_rdm_ep *ep, struct efa_rdm_ope *rxe
 	err = ofi_iov_locate(rxe->iov, rxe->iov_count, ep->msg_prefix_size, &rx_iov_index, &rx_iov_offset);
 	if (OFI_UNLIKELY(err)) {
 		EFA_WARN(FI_LOG_CQ, "ofi_iov_locate failure: %s (%d)\n", fi_strerror(-err), -err);
-		return err;
+		goto err_free;
 	}
+
 	assert(rx_iov_index < rxe->iov_count);
 	assert(rx_iov_offset < rxe->iov[rx_iov_index].iov_len);
 
+	efa_rdm_ope_try_fill_desc(rxe, rx_iov_index, FI_RECV);
+	if (!rxe->desc[rx_iov_index]) {
+		/* efa_rdm_ope_try_fill_desc() did not fill the desc,
+		 * which means memory registration failed.
+		 * return -FI_EAGAIN here will cause user to run progress
+		 * engine, which will cause some memory registration
+		 * in MR cache to be released.
+		 */
+		err = -FI_EAGAIN;
+		goto err_free;
+	}
+
 	pkt_entry->payload = (char *) rxe->iov[rx_iov_index].iov_base + rx_iov_offset;
 	pkt_entry->payload_mr = rxe->desc[rx_iov_index];
 	pkt_entry->payload_size = ofi_total_iov_len(&rxe->iov[rx_iov_index], rxe->iov_count - rx_iov_index) - rx_iov_offset;
 
 	err = efa_rdm_pke_recvv(&pkt_entry, 1);
 	if (OFI_UNLIKELY(err)) {
-		efa_rdm_pke_release_rx(pkt_entry);
 		EFA_WARN(FI_LOG_EP_CTRL,
 			"failed to post user supplied buffer %d (%s)\n", -err,
 			fi_strerror(-err));
-		return err;
+		goto err_free;
 	}
 
 #if ENABLE_DEBUG
 	dlist_insert_tail(&pkt_entry->dbg_entry, &ep->rx_posted_buf_list);
 #endif
 	ep->user_rx_pkts_posted++;
 	return 0;
+
+err_free:
+	if (pkt_entry)
```

Comment:
Is it possible for `pkt_entry` to be `NULL` at this point?

Reply:
Not possible, just for safe in case someone messed the sequence of code in the future. Do you prefer to remove it?